<launch>
  <param name="/use_sim_time" value="true"/>

  <node
      pkg="map_server"
      type="map_server"
      name="map_server"
      args="$(find navigation_stage)/stage_config/maps/willow-full-0.05.pgm 0.05"
      respawn="false"
      />

  <node
      pkg="stage_ros"
      type="stageros"
      name="stageros"
      args="$(find update_move_base_parameter_recovery)/worlds/willow-pr2-5cm.world"
      respawn="false"
      >
    <param name="base_watchdog_timeout" value="0.2"/>
  </node>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
    <param name="footprint_padding" value="0.01" />
    <param name="controller_frequency" value="10.0" />
    <param name="controller_patience" value="3.0" />
    <param name="oscillation_timeout" value="30.0" />
    <param name="oscillation_distance" value="0.5" />

    <rosparam file="$(find update_move_base_parameter_recovery)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find update_move_base_parameter_recovery)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find update_move_base_parameter_recovery)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find update_move_base_parameter_recovery)/config/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find update_move_base_parameter_recovery)/config/dwa_local_planner_params.yaml" command="load" />

    <rosparam>
      base_local_planner: dwa_local_planner/DWAPlannerROS
      recovery_behavior_enabled: true
      recovery_behaviors:
        - name: "update_costmap_complex_local0"
          type: "complex_recovery/SequentialComplexRecovery"
        - name: "conservative_reset"
          type: "clear_costmap_recovery/ClearCostmapRecovery"
        - name: "aggressive_reset"
          type: "clear_costmap_recovery/ClearCostmapRecovery"
        - name: "all_reset"
          type: "clear_costmap_recovery/ClearCostmapRecovery"
      update_costmap_complex_local0:
        recovery_behaviors:
          - name: "update_costmap_local0"
            type: "update_move_base_parameter_recovery/UpdateCostmapParameterRecovery"
          - name: "update_inflation_layer_local0"
            type: "update_move_base_parameter_recovery/UpdateInflationLayerParameterRecovery"
        update_inflation_layer_local0:
          parameter_name: "/move_base_node/local_costmap/inflation_layer"
          inflation_radius: 0.1
          duration_dealline: 9.0
        update_costmap_local0:
          parameter_name: "/move_base_node/local_costmap"
          footprint: "[[-0.34, -0.34], [-0.34, 0.34], [0.34, 0.34], [0.5, 0.0], [0.34, -0.34]]"
          footprint_padding: 0.05
          duration_dealline: 9.0
      conservative_reset:
        reset_distance: 2.0
      aggressive_reset:
        reset_distance: 1.0 # 0.5
      all_reset:
        reset_distance: 0.0
    </rosparam>
  </node>

  <include file="$(find navigation_stage)/move_base_config/amcl_node.xml"/>
  <rosparam>
      /amcl/initial_pose_x: 49.613
      /amcl/initial_pose_y: 29.441
      /amcl/initial_pose_a: 2.245
  </rosparam>

  <node
      name="rviz"
      pkg="rviz"
      type="rviz"
      args="-d $(find update_move_base_parameter_recovery)/config/single_robot.rviz"
      />

</launch>
